aws/resource_aws_vpc_peering_connection_options.go

```diff
@@ -78,14 +79,24 @@ func resourceAwsVpcPeeringConnectionOptionsUpdate(d *schema.ResourceData, meta i
 		return nil
 	}
 
-	pc := pcRaw.(*ec2.VpcPeeringConnection)
-
-	crossRegionPeering := false
-	if aws.StringValue(pc.RequesterVpcInfo.Region) != aws.StringValue(pc.AccepterVpcInfo.Region) {
-		crossRegionPeering = true
-	}
-	if err := resourceAwsVpcPeeringConnectionModifyOptions(d, meta, crossRegionPeering); err != nil {
-		return fmt.Errorf("error modifying VPC Peering Connection (%s) Options: %s", d.Id(), err)
+	if d.HasChanges("accepter", "requester") {
+		pc := pcRaw.(*ec2.VpcPeeringConnection)
+		crossRegionPeering := aws.StringValue(pc.RequesterVpcInfo.Region) != aws.StringValue(pc.AccepterVpcInfo.Region)
+
+		req := &ec2.ModifyVpcPeeringConnectionOptionsInput{
+			VpcPeeringConnectionId: aws.String(d.Id()),
+		}
+		if d.HasChange("accepter") {
+			req.AccepterPeeringConnectionOptions = expandVpcPeeringConnectionOptions(d.Get("accepter").([]interface{}), crossRegionPeering)
+		}
+		if d.HasChange("requester") {
+			req.RequesterPeeringConnectionOptions = expandVpcPeeringConnectionOptions(d.Get("requester").([]interface{}), crossRegionPeering)
+		}
+
+		log.Printf("[DEBUG] Modifying VPC Peering Connection options: %s", req)
```

Comment:
same question regarding waiting here

Reply:
See above.

Reply:
In fact we do have to do some waiting after modification to deal with eventual consistency propagation of acceptance.